src/legacy/core_plugins/data/public/index_patterns/index_patterns_service.mock.ts

```diff
@@ -17,8 +17,51 @@
  * under the License.
  */
 
-import { FieldType } from '../_field';
-import { StaticIndexPattern } from '../_index_pattern';
+// Legacy mocks - must be removed before migrating to new platform.
+jest.mock('ui/chrome');
+
+import { IndexPatternsService, IndexPatternsSetup } from '.';
+import { FieldType } from './fields';
+import { StaticIndexPattern } from './index_patterns';
+
+type IndexPatternsServiceClientContract = PublicMethodsOf<IndexPatternsService>;
+
+const createSetupContractMock = () => {
+  const setupContract: jest.Mocked<IndexPatternsSetup> = {
+    FieldList: {} as any,
+    flattenHitWrapper: jest.fn(),
+    formatHitProvider: jest.fn(),
+    IndexPatterns: jest.fn() as any,
+    ui: {
```

Comment:
I think we're not doing the `ui` section anymore in any other service.
Is there a specific reason why it's required here?

Reply:
yep; currently the `IndexPatternSelect` component uses the saved object client internally, so it isn't static and therefore needs to be provided in the runtime lifecycle contract 

this is something that can easily be refactored out as a follow-up step (along with many other things in index patterns)... i just didn't want to mix concerns inside this PR. 

(though as an aside, i might make the argument for keeping this component as part of the setup contract -- it seems the biggest benefit you get from using it is that its already wired up to the SO client for you and handles making the necessary requests... but that's a discussion for another time)

Reply:
I would prefer to export the UI component normally rather than change the API so frequently. There's no harm in exporting it for the time being in its statefull state (is there?)

All other UI components like SearchBar and QueryBar still have a few stateful deps inside of them. Everything still works, and I feel like doing special exports for such cases is TMI for our users (be in solutions or 3rd party devs).

Reply:
I don't think exporting components statically should be considered "normal" -- if anything it is the inverse: We should assume everything is in the plugin lifecycle contracts, and only export code statically if it is pure (not relying on other services). 

Making a blanket rule to only export UI components statically is going to limit our ability to design components however we'd like.

To go back to the `IndexPatternSelect` example: if all this component is doing is offering a dropdown that's pre-wired to the core Saved Objects client, are we absolutely certain we want to make users pull the Saved Objects client out of their `core` dependencies and pass it as a prop to the component before they can use it? Or do we want to give them the ability to pull in a component and simply use it with an `onChange` callback?

If we forced all UI components to be exported statically, then we wouldn't have the flexibility to make this decision. We'd have no choice but to make the user pass in some stateful dependencies.

Another example: what if we offered a component that was prewired to use state from a different plugin (not core)? If the user of our component wasn't already relying on that plugin, they would now need to add it to the list of their dependencies for the sole purpose of extracting the relevant service from the plugin and passing it to our component. In that case wouldn't it be better to export the component at runtime, already connected to the service?

IMO the more "future-proof" approach is defaulting to exporting contracts at runtime, and only moving to static exports if we are 100% sure the code will never need any state. Otherwise we risk even more confusion for users (in the form of components being moved back and forth between static/runtime contracts).

FWIW, I guess I overlooked the fact that SearchBar & QueryBar still had stateful dependencies; otherwise I would've argued for keeping them in lifecycle contracts until such a time as they could be refactored to become properly stateless.

Reply:
As discussed offline, we'll go ahead and move this to a static export for now, for the sake of consistency, and remove the one chrome reference that lives there. This is not a heavily used component so it should be easy enough to update the interface.

Moving forward, we will look toward a solution for offering "pre-wired" components in runtime contracts, with complementary static components exported statically.